pyo3-build-config/src/impl_.rs

```diff
@@ -155,6 +157,24 @@ impl InterpreterConfig {
         for flag in &self.build_flags.0 {
             println!("cargo:rustc-cfg=py_sys_config=\"{}\"", flag)
         }
+
+        if cfg!(feature = "export-config") {
+            let output_path =
+                Path::new(&cargo_env_var("OUT_DIR").unwrap()).join(EXPORT_CONFIG_FILENAME);
+            if let Ok(config_file) = File::create(&output_path) {
+                if self.to_writer(config_file).is_err() {
+                    warn!(
+                        "Failed to export build config - this may be a problem for external crates that \
+                        depend on the pyo3 config."
+                    );
+                } else {
+                    println!(
+                        "cargo:PYO3_EXPORT_CONFIG={}",
```

Comment:
Rather than relying on the filesystem, should we just dump the config directly to cargo here? Only complexity I can think of is that we might need to escape newlines.

A nice thing about not touching the filesystem is that it's then simple enough that we probably can always have this functionality, rather than needing to enable it with a feature?

Reply:
Yes I think that could work. It seems a bit nice to have the artifact in the filesystem for easy inspection, but there are surely tools/methods to get that data from cargo as well.

As an alternative to dealing with newlines, the data could be put into separate `DEP_PYTHON_<VAR>` env variables so they wouldn't need to be parsed out separately.